pkg-tests/pkg-tests-specs/sources/basic.js

```diff
@@ -0,0 +1,224 @@
+const {tests: {getPackageArchivePath, getPackageHttpArchivePath, getPackageDirectoryPath}} = require(`pkg-tests-core`);
+
+module.exports = makeTemporaryEnv => {
+  describe(`Basic tests`, () => {
+    test(
+      `it should correctly install a single dependency that contains no sub-dependencies`,
+      makeTemporaryEnv(
+        {
+          dependencies: {[`no-deps`]: `1.0.0`},
+        },
+        async ({path, run, source}) => {
+          await run(`install`);
+
+          await expect(source(`require('no-deps')`)).resolves.toMatchObject({
+            name: `no-deps`,
+            version: `1.0.0`,
+          });
+        },
+      ),
+    );
+
+    test(
+      `it should correctly install a dependency that itself contains a fixed dependency`,
+      makeTemporaryEnv(
+        {
+          dependencies: {[`one-fixed-dep`]: `1.0.0`},
+        },
+        async ({path, run, source}) => {
+          await run(`install`);
+
+          await expect(source(`require('one-fixed-dep')`)).resolves.toMatchObject({
+            name: `one-fixed-dep`,
+            version: `1.0.0`,
+            dependencies: {
+              [`no-deps`]: {
+                name: `no-deps`,
+                version: `1.0.0`,
+              },
+            },
+          });
+        },
+      ),
+    );
+
+    test(
+      `it should correctly install a dependency that itself contains a range dependency`,
+      makeTemporaryEnv(
+        {
+          dependencies: {[`one-range-dep`]: `1.0.0`},
+        },
+        async ({path, run, source}) => {
+          await run(`install`);
+
+          await expect(source(`require('one-range-dep')`)).resolves.toMatchObject({
+            name: `one-range-dep`,
+            version: `1.0.0`,
+            dependencies: {
+              [`no-deps`]: {
+                name: `no-deps`,
+                version: `1.1.0`,
+              },
+            },
+          });
+        },
+      ),
+    );
+
+    test(
+      `it should correctly install an inter-dependency loop`,
+      makeTemporaryEnv(
+        {
+          dependencies: {[`dep-loop-entry`]: `1.0.0`},
+        },
+        async ({path, run, source}) => {
+          await run(`install`);
+
+          await expect(
+            source(
+              // eslint-disable-next-line
+              `require('dep-loop-entry') === require('dep-loop-entry').dependencies['dep-loop-exit'].dependencies['dep-loop-entry']`,
+            ),
+          );
```

Comment:
what do you expect here? Missing comparison! If you use eslint-plugin-jest, this will error.

Reply:
Good catch! I ran eslint from the repo root, and errors were successfully reported, but not this one. Maybe because of the await?